src/core_plugins/kibana/public/management/sections/indices/_edit.js

```diff
@@ -84,6 +84,7 @@ uiModules.get('apps/management')
     ingest.delete($scope.indexPattern.id)
     .then($scope.indexPattern.destroy.bind($scope.indexPattern))
     .then(function () {
+      refreshKibanaIndex();
       $location.url('/management/data/index');
     })
```

Comment:
The race condition still exists in this code, because we're not waiting on the promise from `refreshKibanaIndex` to resolve before changing the route. To do this, we need to change the code to:

```
    ingest.delete($scope.indexPattern.id)
    .then($scope.indexPattern.destroy.bind($scope.indexPattern))
    .then(refreshKibanaIndex)
    .then(function () {
      $location.url('/management/data/index');
    })
```